pkg/csi/service/wcp/controller.go

```diff
@@ -541,6 +556,25 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 		log.Error(msg)
 		return nil, err
 	}
+
+	if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.FakeAttach) {
+		fake, err := isFakeAttached(ctx, req.VolumeId)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to determine if volume was fake attached. Error: %v", err)
+			log.Error(msg)
+			return nil, err
+		}
+		if fake {
+			log.Infof("Volume: %s was fake attached", req.VolumeId)
+			//Clear the fake attach annotation
+			annotations := make(map[string]string)
+			annotations[common.AnnFakeAttached] = "no"
```

Comment:
Why dont we remove the annotation instead of changing the value to "no"?

Reply:
Done. Setting the annotation to blank string, which will remove the annotation from pvc in helper method `updatePVCAnnotations`.
```
for key, val := range annotations {
    // If value is not set, remove the annotation
    if val == "" {
            delete(pvcObj.ObjectMeta.Annotations, key)
            log.Debugf("Removing annotation %s on pvc %s/%s", key, pvcNamespace, pvcName)
     }
     .
.
}